pandas/core/indexing.py

```diff
@@ -1550,6 +1551,13 @@ def _setitem_with_indexer(self, indexer, value):
                 val = list(value.values()) if isinstance(value, dict) else value
                 blk = self.obj._mgr.blocks[0]
                 take_split_path = not blk._can_hold_element(val)
+                if isinstance(value, ABCSeries):
```

Comment:
could short_circuit and not do these checks if take_split_path is already True

Reply:
That is a good point, yes.

Reply:
Done